### PR TITLE
Allow applications to download extender config file(s)

### PR DESCRIPTION
**Pull request by fogfish**

### CODE REVIEW

File: api/files/client.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+// New creates a new client instance to fetch config files from PrivX
+func New(api restapi.Connector) *Client {
+	return &Client{api: api}
+}
+
+// ConfigExtender fetches configugartion file
+func (files Client) ConfigExtender(id string) ([]byte, error) {
+	var session struct {
```
Comment (tmozart) [inline api/files/client.go:25]: Shouldn't this be `func (client *Client) ConfigEntender(....)`?

File: api/files/client.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+type Client struct {
+	api restapi.Connector
+}
+
+// New creates a new client instance to fetch config files from PrivX
+func New(api restapi.Connector) *Client {
+	return &Client{api: api}
+}
+
+// ConfigExtender fetches configugartion file
```
Comment (tmozart) [inline api/files/client.go:23]: spelling; configugartion -> configuration.

File: api/files/client.go (hunk 1 above)
Comment (fogfish) [reply api/files/client.go:25]: Sorry. I am not getting this. The component is called `extender` ?

Comment (tmozart) [reply api/files/client.go:25]: sorry for being unclear here... should this method be a Pointer Receiver? It does not alter `files` so it does not have to be, but having pointer receivers seems to be an established convention.

**Pull request merged**